src/Component/Shop/ProductShop/Pagination/index.jsx

```diff
@@ -0,0 +1,90 @@
+import { useSelector, useDispatch } from "react-redux";
+import { useEffect } from "react";
+import {
+  setCurrentPage,
+  getShopProducts,
+  getPagination,
+} from "../../../../store/Slide/ProductsSlide";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faArrowRight, faArrowLeft } from "@fortawesome/free-solid-svg-icons";
+import "./style.scss";
+
+function Pagination({ filter }) {
+  const dispatch = useDispatch();
+  const currentPage = useSelector((state) => state.products.currentPage);
+  const paginations = useSelector((state) => state.products.pagination);
+  const listPage = [];
+  const _page = filter._page;
+
+  useEffect(() => {
+    dispatch(
+      getPagination({
+        ...filter,
+        _page: "",
+        _limit: "",
+      })
+    );
+  }, [filter]);
+
+  if (paginations.list.data) {
+    const allProductsLength = paginations.list.data.length;
+    const listPageLength =
+      allProductsLength % 9
+        ? Math.ceil(allProductsLength / 9)
+        : allProductsLength / 1;
+    for (let i = 1; i <= listPageLength; i++) {
+      listPage.push(i);
+    }
+  }
+
+  const handlePageChange = (num) => {
+    dispatch(getShopProducts({ ...filter, _page: num }));
+    dispatch(setCurrentPage(num));
+  };
+
+  // Xử lý khi có nhiều page
+  const renderPaginationList = () => {
+    const renderBigArr =
+      listPage.length <= 6
+        ? listPage
+        : [_page, _page + 1, _page + 2, "...", listPage.length];
```

Comment:
Check lại logic đoạn này. Giả sử totalPage= 7 currentPage = 5, thì nó sẽ hiện như này
`5, 6, 7, ..., 7`
Ngoài nếu muốn quay lại page 1 thì phải nhấn nút prev nhiều lần mới có thể quay về được
Research về thuật toán pagination